crates/op-rbuilder/src/builders/flashblocks/builder_tx.rs

```diff
@@ -137,96 +137,199 @@ impl BuilderTransactions<FlashblocksExtraCtx, FlashblocksExecutionInfo> for Flas
 // This will be the end of block transaction of a regular block
 #[derive(Debug, Clone)]
 pub(super) struct FlashblocksNumberBuilderTx {
-    pub signer: Option<Signer>,
+    pub signer: Signer,
     pub flashblock_number_address: Address,
+    pub use_permit: bool,
     pub base_builder_tx: BuilderTxBase<FlashblocksExtraCtx>,
     pub flashtestations_builder_tx:
         Option<FlashtestationsBuilderTx<FlashblocksExtraCtx, FlashblocksExecutionInfo>>,
 }
 
 impl FlashblocksNumberBuilderTx {
     pub(super) fn new(
-        signer: Option<Signer>,
+        signer: Signer,
         flashblock_number_address: Address,
+        use_permit: bool,
         flashtestations_builder_tx: Option<
             FlashtestationsBuilderTx<FlashblocksExtraCtx, FlashblocksExecutionInfo>,
         >,
     ) -> Self {
-        let base_builder_tx = BuilderTxBase::new(signer);
+        let base_builder_tx = BuilderTxBase::new(Some(signer));
         Self {
             signer,
             flashblock_number_address,
+            use_permit,
             base_builder_tx,
             flashtestations_builder_tx,
         }
     }
 
-    fn estimate_flashblock_number_tx_gas(
+    fn current_flashblock_number(
```

Comment:
Let's add docs
Will this function just returns current flashblock number? If so why does it needs mutable ref to state?

Reply:
then the read only contract calls only need the call data

Reply:
If you want this to be not mutable -  you need to use `basic_ref` function (that implemented for `DatabaseRef`)